tests/unit/test_vgo_cli_commands.py

```
@@ -359,3 +360,57 @@ def fake_maybe_install_external_dependencies(repo_root: Path, install_mode: str,
     assert 'external_called' not in recorded
     assert recorded['reconcile']['strict_offline'] is True
+
+
+def test_build_parser_exposes_upgrade_subcommand() -> None:
+    parser = build_parser()
+
+    args = parser.parse_args(
+        [
+            'upgrade',
+            '--repo-root',
+            '/tmp/repo',
```

Ruff (0.15.9)

[error] Probable insecure usage of temporary file or directory: "/tmp/repo"

(S108)